libs/wire-subsystems/src/Wire/UserSubsystem/Interpreter.hs

```diff
@@ -527,11 +527,12 @@ guardLockedFields ::
 guardLockedFields user updateOrigin (MkUserProfileUpdate {..}) = do
   let idempName = isNothing name || name == Just user.name
       idempLocale = isNothing locale || locale == user.locale
-      scim = updateOrigin == UpdateOriginWireClient && user.managedBy == Just ManagedByScim
+      scimConflict = updateOrigin == UpdateOriginWireClient && user.managedBy == Just ManagedByScim
+      updateByScim = updateOrigin == UpdateOriginScim && user.managedBy == Just ManagedByScim
```

Comment:
it is always true that `scimConflict == not updateByScim`, so you could replace `not updateByScim` below by `scimConflict`, then you could simplify line 533 below.  and maybe rename `scimConflict` to `updateFromWireClientForScimUser`, or something.

or maybe you could shuffle the code long enough to end up with something involving this:

```
   case updateOrigin of
     Wire... -> _
     Scim... -> _
```

that may be more readable, and would help with refactoring `UpdateOriginType`.

(don't take any of this too seriously, doing this review as a test how my virus-infested brain is recovering.)

Reply:
I gave it a try, but was not happy with the result. I suggest leaving it as is, as I think the readability is good, or you make a suggestion?